library/std/src/sys/unix/process/process_unix.rs

```diff
@@ -695,17 +695,90 @@ impl From<c_int> for ExitStatus {
     }
 }
 
+/// Convert a signal number to a readable, searchable name.
+fn signal_string(signal: i32) -> String {
+    (match signal {
+        libc::SIGHUP => "SIGHUP",
+        libc::SIGINT => "SIGINT",
+        libc::SIGQUIT => "SIGQUIT",
+        libc::SIGILL => "SIGILL",
+        libc::SIGTRAP => "SIGTRAP",
+        libc::SIGABRT => "SIGABRT",
+        libc::SIGBUS => "SIGBUS",
+        libc::SIGFPE => "SIGFPE",
+        libc::SIGKILL => "SIGKILL",
+        libc::SIGUSR1 => "SIGUSR1",
+        libc::SIGSEGV => "SIGSEGV",
+        libc::SIGUSR2 => "SIGUSR2",
+        libc::SIGPIPE => "SIGPIPE",
+        libc::SIGALRM => "SIGALRM",
+        libc::SIGTERM => "SIGTERM",
+        libc::SIGCHLD => "SIGCHLD",
+        libc::SIGCONT => "SIGCONT",
+        libc::SIGSTOP => "SIGSTOP",
+        libc::SIGTSTP => "SIGTSTP",
+        libc::SIGTTIN => "SIGTTIN",
+        libc::SIGTTOU => "SIGTTOU",
+        libc::SIGURG => "SIGURG",
+        libc::SIGXCPU => "SIGXCPU",
+        libc::SIGXFSZ => "SIGXFSZ",
+        libc::SIGVTALRM => "SIGVTALRM",
+        libc::SIGPROF => "SIGPROF",
+        libc::SIGWINCH => "SIGWINCH",
+        libc::SIGIO => "SIGIO",
+        libc::SIGSYS => "SIGSYS",
+        _ => {
+            #[cfg(target_os = "linux")]
+            if signal == libc::SIGSTKFLT {
+                return "SIGSTKFLT".to_string();
+            }
```

Comment:
Rather than shoving these into a catch-all statement, you should just annotate the match arms with these attributes:

```rust
#[cfg(target_os = "linux")]
libc::SIGSTKFLT => "SIGSTKFLT",
```

This should work as-is but you might have to enable a feature flag since I'm not sure if it's stable.

However, doing this will also make sure that the match gets compiled to a single lookup table instead of performing sequential checks.

Reply:
Huh, yeah. That was my first thought, but I knew that didn't used to work. It's allowed in libstd right now, though, so I'll use it.